File: 1732. Find the Highest Altitude.cpp
class Solution {
public:
    int largestAltitude(vector<int>& gain) 
    {
        // Current altitude of the biker
        int current_alti = 0;

        // Highest altitude reached so far
        int maximum_alti = 0;

        // Traverse all altitude changes
        for (int change : gain)
        {
            // Update current altitude
            current_alti += change;

            // Update maximum altitude if needed
            maximum_alti = max(maximum_alti, current_alti);
        }

        return maximum_alti;
    }
};

File: 1732. Find the Highest Altitude.js
/**
 * @param {number[]} gain
 * @return {number}
 */
var largestAltitude = function(gain) {
    // Current altitude of the biker
    let currentAlti = 0;
    
    // Highest altitude reached so far
    let maximumAlti = 0;
    
    // Traverse all altitude changes
    for (let change of gain) {
        // Update current altitude
        currentAlti += change;
        
        // Update maximum altitude if needed
        maximumAlti = Math.max(maximumAlti, currentAlti);
    }
    
    return maximumAlti;
};
